docs/proposals/2170-kubeflow-training-v2/README.md

```diff
@@ -831,8 +830,8 @@ In the future, we can add more parameters if we find use-cases when it is requir
 
 ```golang
 type PodSpecOverride struct {
-	// Names of the training job replicas in the training runtime template to apply the overrides.
-	TargetReplicatedJobs []string `json:"targetReplicatedJobs"`
+	// TrainJobs is the training job replicas in the training runtime template to apply the overrides.
+	TargetJobs []PodSpecOverrideTargetJob `json:"targetJobs"`
```

Comment:
Why do you want to have PodSpecOverrideTargetJob as a separate struct and not `[]string` ?

Reply:
As I described in this PR description, I'm wondering if we can add rules something like labelSelector in the future. So, I expand this field so that we can extend targetJobs without any API breaks.


> Especially, the new targetJobs allows us to support more fine-grained target selection without API breaks in the future.

Reply:
Can you give me an example, on how we can use it as a labelSelector ?

Reply:
Sure. I could imagine the below selector:

```yaml
selector:
  matchLabels:
    role: Worker
```

This is just an idea, and I did not decide to introduce this selector. But, this would be helpful in the situation where the JobSet has multiple Job templates.

Reply:
@tenzen-y  So the API might look something like this ?
```yaml
targetJobs:
  - selector:
      matchLabels:
        role: Worker
```

Does it make sense for the JobSet given that every ReplicatedJob must have unique name ?

Reply:
Yeah, that is what I wanted to say.


```yaml
spec:
    replicatedJobs:
    - name: worker-with-a100-80
      template:
        labels:
          role: worker
    - name: worker-with-a100-40
      template:
        labels:
          role: worker
    - name: worker-with-h100-80
      template:
        labels:
          role: worker
```

We can imagine the above situations. Then we want to override something to all workers.

Reply:
Again, I did not decide to propose this label selector. But, I want to extend the targetJobs fields for future flexibility.